PumaGrasshopper/ComponentPumaBulkInput.cs

```
@@ -130,27 +130,42 @@ private RuleAttributesMap ParseBulkInputTree(IGH_DataAccess DA, int shapeCount)
+                        if(success)
                         {
-                            if (attribute.IsArray())
-                            {
-                                SetRuleAttributeArray(attribute, pair.Value, ref MM);
-                            }
-                            else if(pair.Value.Length > 0)
-                            {
-                                SetRuleAttribute(attribute, pair.Value[0], ref MM);
-                            } 
-                            else
+                            RuleAttribute attribute = attributesList.Find(attr => attr.mFullName.Equals(pair.Key) || attr.mNickname.Equals(pair.Key));
+
+                            if (attribute != null)
                             {
-                                throw new Exception("Attribute key " + pair.Key + " has no corresponding value");
+                                if (attribute.IsArray())
+                                {
+                                    SetRuleAttributeArray(attribute, pair.Value, ref MM);
+                                }
+                                else if(pair.Value.Length > 0)
+                                {
+                                    SetRuleAttribute(attribute, pair.Value[0], ref MM);
+                                }
+                                else
+                                {
+                                    throw new Exception("Attribute key " + pair.Key + " has no corresponding value");
+                                }
                             }
                         }
-                    });
+                        else if(pair.Key.Length == 0)
+                        {
+                            AddRuntimeMessage(GH_RuntimeMessageLevel.Warning,
+                                              "Branch " + branchId + " - Line " + i + ": Ignoring empty line.");
+                        }
```

I suggest to remove this warning, i.e. silently ignore empty lines.

---

I did the changes you requested. Ready for re-review 👍

---

Looks good